test/integration/scheduler/queue_test.go

```diff
@@ -1345,6 +1353,165 @@ func TestCoreResourceEnqueue(t *testing.T) {
+				}}).Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				csinode1 := st.MakeCSINode().Name("csinode").Obj()
+				if _, err := testCtx.ClientSet.StorageV1().CSINodes().Create(testCtx.Ctx, csinode1, metav1.CreateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to create CSINode: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{{Resource: framework.CSINode, ActionType: framework.Add}: 1}, nil
+			},
+			wantRequeuedPods:          sets.New("pod1"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected with PVC by the CSI plugin is requeued when the pod having related PVC is deleted",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
+			initialPVs: []*v1.PersistentVolume{
+				st.MakePersistentVolume().Name("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					PersistentVolumeSource(v1.PersistentVolumeSource{CSI: &v1.CSIPersistentVolumeSource{Driver: "csidriver", VolumeHandle: "volumehandle1"}}).
+					Obj(),
+				st.MakePersistentVolume().Name("pv2").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					PersistentVolumeSource(v1.PersistentVolumeSource{CSI: &v1.CSIPersistentVolumeSource{Driver: "csidriver", VolumeHandle: "volumehandle2"}}).
+					Obj(),
+			},
+			initialPVCs: []*v1.PersistentVolumeClaim{
+				st.MakePersistentVolumeClaim().
+					Name("pvc1").
+					Annotation(volume.AnnBindCompleted, "true").
+					VolumeName("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOncePod}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}}).
+					Obj(),
+				st.MakePersistentVolumeClaim().
+					Name("pvc2").
+					Annotation(volume.AnnBindCompleted, "true").
+					VolumeName("pv2").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOncePod}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}}).
+					Obj(),
+			},
+			initialCSIDrivers: []*storagev1.CSIDriver{
+				st.MakeCSIDriver().Name("csidriver").StorageCapacity(ptr.To(true)).Obj(),
+			},
+			initialCSINodes: []*storagev1.CSINode{
+				st.MakeCSINode().Name("fake-node").Driver(storagev1.CSINodeDriver{Name: "csidriver", NodeID: "fake-node", Allocatable: &storagev1.VolumeNodeResources{
+					Count: ptr.To(int32(1)),
+				}}).Obj(),
+			},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Node("fake-node").Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod2").Container("image").PVC("pvc2").Obj(),
```

Comment:
ditto, please add another Pod that is rejected by the plugin, but won't be requeued.

Reply:
It's not possible in this case. Pod delete event with _any_ volumes causes requeueing.

Reply:
Yes, that's right.
https://github.com/utam0k/kubernetes/blob/98bcbc857b858983e68c132b5c2de077a0b2e9db/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L110-L115